### PR TITLE
Dynamic data format without specifying data_format in `backtest`

### PR DESCRIPTION
**Pull request by enzoampil**
This PR which should resolve issue #83 allows for the data format of a dataframe used as an input in `backtest` to be automatically inferred based on the column names. Detectable column names can be found as keys in `config.DATA_FORMAT_BASE` (aside from `dt`).

@jpdeleon mind reviewing since we were the ones working this before? :smile:

### CODE REVIEW

**Comment by jpdeleon** [comment]
will do during the meeting!

File: python/fastquant/backtest/backtest.py [hunk 1]
```diff
@@ -68,18 +70,43 @@ def dec(obj):
...
-    data_format="c",
     plot=True,
     verbose=True,
     sort_by="rnorm",
     sentiments=None,
     strats=None,  # Only used when strategy = "multi"
+    data_format=None,  # If none, format is automatically inferred
     **kwargs
 ):
     """
```
Comment (jpdeleon) [inline python/fastquant/backtest/backtest.py:112]: add docstring
```
Parameters
----------------
strategy : str 
    see list of accepted strategy keys below
data : pandas.DataFrame
    dataframe with at least close price indexed with time
commission : float
    commission per transaction [0, 1]
init_cash : float
    initial cash (currency implied from `data`)
plot : bool
    show plot backtrader (disabled if `strategy`=="multi")
sort_by : str
    sort result by given metric (default='rnorm')
sentiments : pandas.DataFrame
    df of sentiment [0, 1] indexed by time (applicable if `strategy`=='senti')
strats : dict
    dictionary of strategy parameters (applicable if `strategy`=='multi')
data_format : str
    data e.g. ohlcv (default=None so format is automatically inferred)
```

Comment (enzoampil) [reply python/fastquant/backtest/backtest.py]: Awesome added!

**Comment by enzoampil** [comment]
Awesome thanks for the review @jpdeleon !!

**Pull request merged**